### PR TITLE
Fix SubtractResourcesStrict if requested resource is 0

### PR DESCRIPTION
**Pull request by pcmoritz**
This was encountered in https://github.com/ray-project/ray/issues/4870

Feels unclean, there should be a better way to do this. Any suggestions are welcome.

@romilbhardwaj @robertnishihara

### CODE REVIEW

File: src/ray/raylet/scheduling_resources.cc [hunk 1]
```diff
@@ -168,6 +168,13 @@ void ResourceSet::SubtractResourcesStrict(const ResourceSet &other) {
   for (const auto &resource_pair : other.GetResourceAmountMap()) {
     const std::string &resource_label = resource_pair.first;
     const FractionalResourceQuantity &resource_capacity = resource_pair.second;
+
+    // It is valid to request 0 resources from a non-existent resource,
+    // so if that happens, we skip all the checking logic and updating below.
+    if (resource_capacity == 0) {
```
Comment (robertnishihara) [inline src/ray/raylet/scheduling_resources.cc]: I think this is a different bug. I thought we maintain as an inariant that `ResourceSet` has no resources with capacity 0.

**Comment by robertnishihara** [comment]
@pcmoritz please add a test that triggers the issue.

**Comment by pcmoritz** [comment]
It is possible that the zero resource sneaked in via https://github.com/ray-project/ray/blob/master/src/ray/raylet/scheduling_resources.cc#L77

I haven't been able to reproduce it yet btw.

**Comment by robertnishihara** [comment]
cc @williamma12 would you be able to look into this?

**Comment by pcmoritz** [comment]
this is related: https://github.com/ray-project/ray/issues/4892

**Comment by pcmoritz** [comment]
already fixed on master now

**Pull request closed without merging**